keycloak4s-auth/core/src/main/scala/com/fullfacing/keycloak4s/auth/core/PayloadImplicits.scala

```diff
@@ -14,15 +15,15 @@ object PayloadImplicits {
 
   /* A safe extraction method to extract any field's value from a Payload. **/
   private def safeExtract(payload: Payload, key: String): Option[String] = Option {
-    payload.toJSONObject.getAsString(key)
+    new JSONObject(payload.toJSONObject).getAsString(key)
   }
 
   implicit class PayloadImpl(payload: Payload) {
 
     /* Generic Extractors. **/
 
     def extract(key: String): Option[String] =
-      safeExtract(payload, key)
+      safeExtract(payload, key).map(read[String])
```

Comment:
Why is the `read` necessary here?

Reply:
Ah, code I added and forgot to remove. Was having issues with the API as nimbus shaded the Json Smart library they were using.

Reply:
Resolved.